### PR TITLE
Use database lock for distributed locking in Event generator

### PR DESCRIPTION
**Pull request by paltanmoy**
#145999281

### CODE REVIEW

File: src/autoscaler/eventgenerator/cmd/eventgenerator/main.go [hunk 1]
```diff
@@ -111,22 +113,43 @@ func main() {
...
 
+	guid, err := generateGUID(logger)
+	if err != nil {
+		logger.Error("failed-to-generate-guid", err)
+	}
+
+	if conf.EnableDBLock {
+		logger.Debug("database-lock-feature-enabled")
+		var lockDB db.LockDB
+		var lockTableName = "eg_lock"
```
Comment (boyang9527) [inline src/autoscaler/eventgenerator/cmd/eventgenerator/main.go]: define it as a const

File: src/autoscaler/eventgenerator/cmd/eventgenerator/main.go [hunk 2]
```diff
@@ -260,10 +283,11 @@ func initializeRegistrationRunner(
 	return locket.NewRegistrationRunner(logger, registration, consulClient, locket.RetryInterval, clock)
 }
 
-func generateGUID(logger lager.Logger) string {
-	uuid, err := uuid.NewV4()
+func generateGUID(logger lager.Logger) (string, error) {
```
Comment (boyang9527) [inline src/autoscaler/eventgenerator/cmd/eventgenerator/main.go]: move this function out of eventGen, make it shared among components

**Comment by paltanmoy** [comment]
@boyang9527 : Done

**Comment by boyang9527** [comment]
@paltanmoy  please use the common util in metrics collector too.

**Comment by paltanmoy** [comment]
@boyang9527 updated the PR.

**Pull request merged**